src/Engine/EmitMSIL/Replication.cs

```diff
@@ -210,6 +210,10 @@ public static object ReplicationLogic(List<CLRFunctionEndPoint> feps, IList args
             object result;
             if (replicationInstructions.Count == 0)
             {
+                // TODO: Ideally, we should not reach here as no-replication cases should ideally be all handled
+                // at compile-time. Uncomment the following Validity.Assert to debug these remaining cases.
+                //Validity.Assert(replicationInstructions.Count != 0,
+                //    "No-replication case, not expected to be handled inside ReplicationLogic function.");
```

Comment:
@aparajit-pratap nice - should this be a jira task?

Reply:
Will file it.